scanner/redhatbase.go

```diff
@@ -110,6 +103,33 @@ func detectRedhat(c config.ServerInfo) (bool, osTypeInterface) {
 		}
 	}
 
+	// https://bugzilla.redhat.com/show_bug.cgi?id=1332025
+	// CentOS cloud image
+	if r := exec(c, "ls /etc/centos-release", noSudo); r.isSuccess() {
+		if r := exec(c, "cat /etc/centos-release", noSudo); r.isSuccess() {
+			re := regexp.MustCompile(`(.*) release (\d[\d\.]*)`)
+			result := re.FindStringSubmatch(strings.TrimSpace(r.Stdout))
+			if len(result) != 3 {
+				logging.Log.Warnf("Failed to parse CentOS version: %s", r)
+				return true, newCentOS(c)
+			}
+
+			release := result[2]
+			switch strings.ToLower(result[1]) {
+			case "centos", "centos linux", "centos stream":
+				cent := newCentOS(c)
+				cent.setDistro(constant.CentOS, release)
+				return true, cent
+			case "alma", "almalinux":
+				alma := newAlma(c)
+				alma.setDistro(constant.Alma, release)
+				return true, alma
```

Comment:
This may not be necessary, since the AlmaLinux decision is made before here.

Reply:
How About Line 160?
You may be right, But It seems  better to be careful.

Reply:
Then you should also consider Rocky Linux from `/etc/centos-release`. 
Also, consider Fedora from `/etc/redhat-release`. 
```diff
:100644 100644 9288014 0000000 M	scanner/redhatbase.go

diff --git a/scanner/redhatbase.go b/scanner/redhatbase.go
index 9288014..f341171 100644
--- a/scanner/redhatbase.go
+++ b/scanner/redhatbase.go
@@ -121,6 +121,10 @@ func detectRedhat(c config.ServerInfo) (bool, osTypeInterface) {
 				alma := newAlma(c)
 				alma.setDistro(constant.Alma, release)
 				return true, alma
+			case "rocky", "rocky linux":
+				rocky := newRocky(c)
+				rocky.setDistro(constant.Rocky, release)
+				return true, rocky
 			default:
 				logging.Log.Warnf("Failed to parse CentOS: %s", r)
 			}
@@ -149,6 +153,10 @@ func detectRedhat(c config.ServerInfo) (bool, osTypeInterface) {
 				logging.Log.Warnf("Versions prior to RedHat/CentOS 5 are not supported, detected version is %s", release)
 			}
 			switch strings.ToLower(result[1]) {
+			case "fedora":
+				fed := newFedora(c)
+				fed.setDistro(constant.Fedora, release)
+				return true, fed
 			case "centos", "centos linux", "centos stream":
 				cent := newCentOS(c)
 				cent.setDistro(constant.CentOS, release)

```